mlflow/store/tracking/dbmodels/models.py

```diff
@@ -1271,3 +1272,225 @@ class SqlLoggedModelTag(Base):
+
+    dataset_id = Column(String(36), primary_key=True)
+    """
+    Dataset ID: `String` (limit 36 characters).
+    *Primary Key* for ``evaluation_datasets`` table.
+    """
+
+    name = Column(String(255), nullable=False)
+    """
+    Dataset name: `String` (limit 255 characters). *Non null* in table schema.
+    """
+
+    source = Column(String(255), nullable=True)
+    """
+    Dataset source: `String` (limit 255 characters).
+    """
+
+    source_type = Column(String(50), nullable=True)
+    """
+    Source type: `String` (limit 50 characters).
+    """
+
+    schema = Column(Text, nullable=True)
+    """
+    Schema information: `Text`.
+    """
+
+    profile = Column(Text, nullable=True)
+    """
+    Profile information: `Text`.
+    """
+
+    digest = Column(String(64), nullable=True)
+    """
+    Dataset digest: `String` (limit 64 characters).
+    """
+
+    created_time = Column(BigInteger, default=get_current_time_millis)
+    """
+    Creation time: `BigInteger`.
+    """
+
+    last_update_time = Column(BigInteger, default=get_current_time_millis)
+    """
+    Last update time: `BigInteger`.
+    """
+
+    created_by = Column(String(255), nullable=True)
+    """
+    Creator user ID: `String` (limit 255 characters).
+    """
+
+    last_updated_by = Column(String(255), nullable=True)
+    """
+    Last updater user ID: `String` (limit 255 characters).
+    """
+
+    # Relationship to records
+    records = relationship(
+        "SqlEvaluationDatasetRecord", back_populates="dataset", cascade="all, delete-orphan"
+    )
+
+    __table_args__ = (
+        PrimaryKeyConstraint("dataset_id", name="evaluation_datasets_pk"),
+        Index("index_evaluation_datasets_name", "name"),
+        Index("index_evaluation_datasets_created_time", "created_time"),
+    )
+
+
+class SqlEvaluationDatasetRecord(Base):
+    """
+    DB model for evaluation dataset records.
+    """
+
+    __tablename__ = "evaluation_dataset_records"
+
+    dataset_record_id = Column(String(36), primary_key=True)
+    """
+    Dataset record ID: `String` (limit 36 characters).
+    *Primary Key* for ``evaluation_dataset_records`` table.
+    """
+
+    dataset_id = Column(
+        String(36), ForeignKey("evaluation_datasets.dataset_id", ondelete="CASCADE"), nullable=False
+    )
+    """
+    Dataset ID: `String` (limit 36 characters). Foreign key to evaluation_datasets.
+    """
+
+    inputs = Column(Text, nullable=False)
+    """
+    Inputs JSON: `Text`. *Non null* in table schema.
+    """
+
+    expectations = Column(Text, nullable=True)
+    """
+    Expectations JSON: `Text`.
+    """
+
+    tags = Column(Text, nullable=True)
+    """
+    Tags JSON: `Text`.
+    """
+
+    source = Column(Text, nullable=True)
+    """
+    Source JSON: `Text`.
+    """
+
+    source_id = Column(String(36), nullable=True)
+    """
+    Source ID for lookups: `String` (limit 36 characters).
+    """
+
+    source_type = Column(Text, nullable=True)
```

Comment:
can `source_type` be long?

Reply:
I was originally concerned that this might need to be flexible enough for user-specified source type information, but I feel like we should just lock this down to support enum controlled values only. I'm going to move this to a String type capped at 255 characters to reduce table allocation size.